wkhtmltopdf/utils.py

```diff
@@ -231,14 +235,22 @@ def make_absolute_paths(content):
 
     return content
 
-def render_to_temporary_file(template, context, mode='w+b', bufsize=-1,
-                                 suffix='.html', prefix='tmp', dir=None,
-                                 delete=True):
-    # make sure the context is a context object
-    if not isinstance(context, (Context, RequestContext)):
-        context = Context(context)
+def render_to_temporary_file(template, context, request=None, mode='w+b',
+                             bufsize=-1, suffix='.html', prefix='tmp',
+                             dir=None, delete=True):
+    if django.VERSION < (1, 8):
+        # If using a version of Django prior to 1.8, ensure ``context`` is an
+        # instance of ``Context``
+        if not isinstance(context, Context):
+            if request:
+                context = RequestContext(request, context)
+            else:
+                context = Context(context)
+        content = template.render(context)
+    else:
+        content = template.render(context, request)
```

Comment:
Unfortunately, it's not really that obvious.

The `Template` class you're looking at is internal to the Django template library. In the above code, when Django 1.7 is being used, `template` is an instance of the class you pointed out above.

In 1.8, different template languages backends ("engines") can be used and they often have their own `Template` class. Django's (DTL for short) is now just one of them. [Jinja2 has another one](https://github.com/mitsuhiko/jinja2/blob/master/jinja2/environment.py#L871) and Django supports it out of the box.

The signature of the `render()` method for each, however, is a bit different. DTL's is `Template.render(self, context)` and Jinja2's is def `Template.render(self, *args, **kwargs)`.

To provide a uniform interface, the methods on each engine's that return templates return an instance of a *different* `Template` class that is just required to have one method: `Template.render(self, context=None, request=None)`. [Here is DTL's](https://github.com/django/django/blob/master/django/template/backends/django.py#L63) and [here is Jinja2's](https://github.com/django/django/blob/master/django/template/backends/jinja2.py#L64).